core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -220,6 +220,16 @@ private static String toSql(RelNode root, SqlDialect dialect,
         .getSql();
   }
 
+  /**
+   * Test for <a href="https://issues.apache.org/jira/browse/CALCITE-5988">[CALCITE-5988]</a>
+   * SqlImplementor.toSql cannot emit VARBINARY literals.
+   */
+  @Test void testBinary() {
+    String query = "SELECT x'ABCD'";
+    String expected = "SELECT X'ABCD'";
+    sql(query).withMysql().ok(expected);
```

Comment:
Is `withMysql` necessary to reproduce the issue, if not, I'd suggest to just use `sql(query).ok(exptected)`

Reply:
I have added a comment explaining this choice, but if you prefer to use the default Calcite dialect I can change the test.

Reply:
I'm fine with leaving as it is.